presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -148,7 +148,7 @@ public static PlanNodeStatsEstimate intersect(PlanNodeStatsEstimate left, PlanNo
                             .setNullsFraction(min(leftSymbolStats.getNullsFraction(), rightSymbolStats.getNullsFraction()))
                             .build());
         }
-
+        statsBuilder.setOutputRowCount(left.getOutputRowCount() + right.getOutputRowCount());  // this is the maximum row count;
```

Comment:
It was caught by unit tests for TestIntersectStatsRule: https://travis-ci.org/Teradata/presto/jobs/259865594 .